### ISSUE TITLE
Hasura schema-sync-thread error

### ISSUE THREAD

**Issue by mikevb3** [hasura/graphql-engine#6718]
Hasura Version: 1.3.3, migrated from 1.3.2
GKE

This past week we've been having downtimes due to introspection causing a connection error. Checking the Kubernetes pod, we got this message.

`
{
  "type":"schema-sync-thread",
  "timestamp":"2021-03-25T23:25:17.885+0000",
  "level":"error",
  "detail":{
    "thread_type":"listener",
    "info":{
      "error":{
        "type":"query_error",
        "info":{
          "internal":"server closed the connection unexpectedly\n\tThis probably means the server terminated abnormally\n\tbefore or while processing the request.\n",
          "path":"$",
          "error":"connection error",
          "code":"postgres-error"
        }
      }
    }
  }
}`

And when executing the get inconsistent metadata endpoint, causes a 502 error. Anyone have had this kind of error?

**Comment by tirumaraiselvan** [comment]
Is there anything in Postgres logs, the issue is seemingly coming from postgres.

**Comment by mikevb3** [comment]
Thank you @tirumaraiselvan , at the moment i changed jobs, and i'm not longer at the startup where the incident began. but i will gladly share you what we found out.

Apparently we had a really large metadata, and it was causing some introspection errors, and Postgres indeed shutdown itself. it was a cloud sql gcp instance, but we did a clean install with a copy of the production db, on hasura cloud, and it worked well.
